erpnext/stock/doctype/stock_entry_detail/stock_entry_detail.json

```diff
@@ -346,7 +346,8 @@
    "fieldtype": "Check",
    "label": "Allow Zero Valuation Rate",
    "no_copy": 1,
-   "print_hide": 1
+   "print_hide": 1,
+   "read_only_depends_on": "eval: doc.s_warehouse"
```

Comment:
Tested locally and found one more UX problem.

You can toggle the checkbox and then select s_warehouse, then it will have an incorrect value AND won't let you change it. 

Should be fixable by setting the checkbox to zero whenever the source warehouse is set.